torch/_dynamo/config.py

```diff
@@ -322,6 +322,9 @@ def _get_optimize_ddp_mode():
 # use numpy's PRNG if True, pytorch otherwise
 use_numpy_random_stream = False
 
+# Use C++ guard manager
+enable_cpp_guard_manager = True
```

Comment:
I will set this to False before merging the PR. After some internal testing, I will add another PR to turn this ON.